### PR TITLE
feat: new RadioButton component, #171

### LINKED ISSUE #171: Create custom radio button component
Our design system is currently missing a `RadioButton` component that blends in with the theme of the site in the same way our [`Checkbox`](https://5app.github.io/base5-ui/?path=/docs/components-checkbox--default-story) component does.

We should create one that works just the same:
- API should mirror the native `<input type="radio" />`
- Like `Checkbox`, it should have an additional `scale` prop with the same behaviour
- Its size & layout properties should match that of the `Checkbox` component as well, so they can be swapped out for each other seamlessly
- It's design should follow the default mac OS radio buttons, and the size & colours of the `Checkbox` component:
  - A simple outlined circle when unselected
  - A filled in circle when selected (`theme.links` colour), and instead of a checkmark icon, a smaller, white outlined circle in the center
  - Cross-hatched filling when disabled (though it may make sense to revisit this design in both the `Checkbox` component and here – maybe a simple grey filling is a better look)
  - Thick focus indicator outline (`theme.links` colour) when using a keyboard

Example of default mac OS checkbox design:
image

### PR DESCRIPTION
**Pull request by SarahHouben**
fixes #171 

Adds the RadioButton component.
I've attempted to stay as close to the implementation to the Checkbox as possible, however the components might not have 100% the same scaling behaviour.

Bonus: updates a few dependencies and adds `eslint-plugin-storybook`

Screenshot 2021-12-23 at 17 11 03

### CODE REVIEW

File: src/RadioButton/index.js [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+			if (onClick) onClick(e);
+
+			/**
+			 * IE 11 and Edge don't fire onChange for indeterminate checkboxes,
+			 * therefore we're (ab)using the click handler to do so instead.
+			 *
+			 * Surprisingly, this even works with keyboard navigation
+			 * (i.e. when pressing Space when the input is focused).
+			 */
+			if (indeterminate && onChange) {
```
Comment (SarahHouben) [inline src/RadioButton/index.js:159]: Not sure whether we would ever use the `indeterminate` option for RadioButtons but having it seemed to not hurt

File: src/RadioButton/index.js [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+	background: ${p => contrast(p.theme.links)};
+	border: 1px solid
+		${p =>
+			alpha(
+				p.isChecked ? p.theme.links : p.theme.text,
+				p.isChecked ? 1 : 0.5
+			)};
+	border-radius: ${pxToRem(16)};
+
+	input[disabled] + & {
```
Comment (SarahHouben) [inline src/RadioButton/index.js:59]: Not sure whether the disabled `RadioButtons` should maybe also have an outline. I've left it for now in order to be more consistent with the `Checkbox` UI.

`Checkbox` in comparison:
Screenshot 2022-01-04 at 11 37 15

File: .eslintrc.js [hunk 3]
```diff
@@ -1,67 +1,59 @@
...
-			},
-		},
-	],
-	settings: {
-		react: {
-			version: 'detect',
-		},
-	},
-};
+  plugins: ['react-hooks', 'jsx-a11y', '@typescript-eslint'],
```
Comment (MrSwitch) [inline .eslintrc.js:2]: The formatting looks pretty bad here? I presume this was done by prettier, can we create a rule to remove it in the PR?

**Review by MrSwitch — APPROVED** [review]
Could you please revert the changes to .eslintrc.js, otherwise great 👍

**Review by MrSwitch — APPROVED** [review]

File: src/RadioButton/index.js (hunk 2 above)
Comment (diondiondion) [reply src/RadioButton/index.js:59]: Looks fine to me in the screenshot 👍

File: src/RadioButton/index.js (hunk 1 above)
Comment (diondiondion) [reply src/RadioButton/index.js:159]: I think it'd be best to remove the indeterminate state – it does look nice, but indeterminate radio buttons aren't really a thing and run the risk of being confusing and inaccessible. It also simplifies things a bit in this component – you don't need the ref merging and useCallback.

File: src/RadioButton/index.js [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+				tabIndex={tabIndex || tabIndexContext}
+				{...otherProps}
+			/>
+			<MockBox
+				aria-hidden="true"
+				isChecked={isChecked || indeterminate}
+				scale={scale}
+			>
+				<Icon
+					name={indeterminate ? 'minus' : 'dot'}
```
Comment (diondiondion) [inline src/RadioButton/index.js:186]: I'm surprised the existing "dot" icon works so well here! I don't even remember what the dot icon is used for outside of this component… 🤔

**Review by diondiondion — APPROVED** [review]
Thanks, this looks great, and matches the Checkbox component. 👍  I'd just remove the indeterminate state, otherwise this is good to go.

**Pull request merged**